modules/javafx.graphics/src/main/resources/com/sun/prism/es2/glsl/main.vert

```diff
@@ -37,12 +37,17 @@ struct Light {
     vec3 color;
     vec3 attn;
     float range;
+    vec3 dir;
+    float cosOuter;
+    float denom; // cosInner - cosOuter
+    float falloff;
 };
 
 //3 lights used
 uniform Light lights[3];
 
 varying vec4 lightTangentSpacePositions[3];
+varying vec4 lightTangentSpaceDirections[3];
```

Comment:
Since this is preexisting, I probably wouldn't change it as part of this PR, but instead it could be done as part of the future work to support more than 3 lights.